### PR TITLE
Tweak elasticsearch-env script to work under posix

### LINKED ISSUE #65980: Shell scripts fail under POSIXLY_CORRECT
Trying to follow https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-security.html .
Unfortunately elasticsearch-setup-passwords (elasticsearch-env) seems to be broken:
```
/usr/share/elasticsearch/bin/elasticsearch-setup-passwords interactive
/usr/share/elasticsearch/bin/elasticsearch-env: line 126: syntax error near unexpected token `<'
```

So how to secure elasticsearch?

### LINKED ISSUE #66656: [CI] DockerTests test085EnvironmentVariablesAreRespectedUnderDockerExec failing on Fedora 32
This docker test seems to fail on occasion specifically on Fedora 32:

https://gradle-enterprise.elastic.co/s/zjf65ovw657li/tests/:qa:os:destructiveDistroTest.default-docker_ubi/org.elasticsearch.packaging.test.DockerTests/test085EnvironmentVariablesAreRespectedUnderDockerExec#1

```
org.elasticsearch.packaging.test.DockerTests > test010Install FAILED
    java.lang.RuntimeException: Command was not successful: [bash -c docker exec --user elasticsearch:root --tty 2e1612134c884d03d13f7a711a416b26db7e2d0c5643c19aa9a998305f313fe1 stat --format="%U %G %A" /usr/share/elasticsearch/config/elasticsearch.keystore]
       result: exitCode = [1] stdout = [stat: cannot stat '/usr/share/elasticsearch/config/elasticsearch.keystore': No such file or directory] stderr = []
        at __randomizedtesting.SeedInfo.seed([528C4E8802F598B8:76D3EE1B6058C7E0]:0)
        at org.elasticsearch.packaging.util.Shell.runScript(Shell.java:158)
        at org.elasticsearch.packaging.util.Shell.run(Shell.java:84)
        at org.elasticsearch.packaging.util.Docker.assertPermissionsAndOwnership(Docker.java:372)
        at org.elasticsearch.packaging.util.Docker.verifyOssInstallation(Docker.java:426)
        at org.elasticsearch.packaging.util.Docker.verifyContainerInstallation(Docker.java:408)
        at org.elasticsearch.packaging.test.DockerTests.test010Install(DockerTests.java:114)
```

### LINKED ISSUE #67097: [CI] DockerTests.test085EnvironmentVariablesAreRespectedUnderDockerExec fails with missing keystore file
This test `DockerTests.test085EnvironmentVariablesAreRespectedUnderDockerExec` fails everyday on master, 7.11, 7.10 or 7.x. Another issue #66656 is opened for this test but the error is different so I'm opening a new issue.

**Build scans**:
https://gradle-enterprise.elastic.co/s/jkuoddwnkjkzs
https://gradle-enterprise.elastic.co/s/pxr65mf6xmhqs
https://gradle-enterprise.elastic.co/s/gy5sbg4msiycw

**Applicable branches**:
master, 7.x, 7.11 and 7.10

**Failure history**:
Build history shows [many failures](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:'2020-11-07T13:31:46.218Z',mode:absolute,to:'2021-01-06T13:31:46.220Z'))&_a=(columns:!(branch,test),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'test:%22test085EnvironmentVariablesAreRespectedUnderDockerExec%22'),sort:!(time,desc))) over the last month.

**Failure excerpt**:
```
org.elasticsearch.packaging.test.DockerTests > test085EnvironmentVariablesAreRespectedUnderDockerExec FAILED
    java.lang.AssertionError: 
    Expected: a string containing "java.net.UnknownHostException: this.is.not.valid: Name or service not known"
         but: was "ERROR: Elasticsearch keystore file is missing [/usr/share/elasticsearch/config/elasticsearch.keystore]"
        at __randomizedtesting.SeedInfo.seed([23D04474A3D733E1:857F2D053C98986E]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.packaging.test.DockerTests.test085EnvironmentVariablesAreRespectedUnderDockerExec(DockerTests.java:486)
```

### PR DESCRIPTION
**Pull request by pugnascotia**
Closes #65980. Also closes #67097 and closes #66656.

The elasticsearch-env script used a bash feature called "process
substitution", which doesn't work when `POSIXLY_CORRECT` is set in the
environment (the construct doesn't even parse). Tweak the implementation
to use something that works under `POSIXLY_CORRECT`.

Also add a test to `ArchiveTests`, which fails without this fix, and
rework `DockerTests.test085EnvironmentVariablesAreRespectedUnderDockerExec`
since a couple of issues were open against it.

### CODE REVIEW

File: qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java [hunk 1]
```diff
@@ -476,16 +480,23 @@ public void test084SymlinkToFileWithInvalidPermissionsIsRejected() throws Except
      * Check that environment variables are translated to -E options even for commands invoked under
      * `docker exec`, where the Docker image's entrypoint is not executed.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/67097")
-    public void test085EnvironmentVariablesAreRespectedUnderDockerExec() {
+    public void test085EnvironmentVariablesAreRespectedUnderDockerExec() throws Exception {
```
Comment (mark-vieira) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java:483]: So does this address #67097 as well?

**Review by mark-vieira — APPROVED** [review]
Awesome! Thanks for investigating this, Rory. LGTM.

File: qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java (hunk 1 above)
Comment (pugnascotia) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java:483]: Yes, there's 2 issues open against this test and I'm hoping this change resolves both.

Comment (mark-vieira) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java:483]: Should we add them to the "closes" description of this issue?

Comment (pugnascotia) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java:483]: Gah, I knew, I _knew_, I'd forget to add those 🤦

**Comment by pugnascotia** [comment]
Backports:
   - `7.x` in 29ba82e3a2d (and a fixup in 0ff9b321aa6)
   - `7.11` in a10c533b5dc
   - `7.10` in bda21b2b153

**Pull request merged**